themes/classic/_dev/js/components/product-miniature.js

```diff
@@ -28,24 +28,34 @@ export default class ProductMinitature {
   init() {
     $('.js-product-miniature').each((index, element) => {
       const FLAG_MARGIN = 10;
-      const discountElems = $(element).find('.discount-product');
+      //Top on sale banner
       const onSaleElems =  $(element).find('.on-sale');
-      const newElems = $(element).find('.new');
-
-      if (discountElems.length) {
-        newElems.css('top', discountElems.height() * 3 + FLAG_MARGIN);
-        discountElems.css('top', -$(element).find('.thumbnail-container').height() + $(element).find('.product-description').height() + FLAG_MARGIN);
+      //Discount flag
+      const discountElems = $(element).find('.discount-product');
+      //Flags other than on-sale, discount and online-only (which have all their way to display)
+      const flagElems = $(element).find('.product-flag:not(.on-sale):not(.discount):not(.online-only)');
 
-        if ($(element).find('.pack').length) {
-          $(element).find('.pack').css('top', discountElems.height() * 3 + FLAG_MARGIN);
-        }
+      var flagsTop = FLAG_MARGIN;
+      var discountTop = FLAG_MARGIN;
+      if (onSaleElems.length) {
+        discountTop = parseFloat(onSaleElems.outerHeight()) + FLAG_MARGIN;
```

Comment:
We're playing with pixel, better to 
```suggestion
        discountTop = parseInt(onSaleElems.outerHeight(), 10) + FLAG_MARGIN;
```

Reply:
Actually it's a rest of former code, I think we can safely remove the `parse*` part